src/pages/Login.tsx

```diff
@@ -0,0 +1,225 @@
+                                        required
+                                        fullWidth
+                                        autoComplete="email"
+                                        autoFocus
+                                        color="secondary"
+                                        error={!!errors.email}
+                                        helperText={errors.email ? errors.email.message : ''}
+                                    />
+                                )}
+                            />
+
+                            <Controller
+                                name="password"
+                                control={control}
+                                defaultValue=""
+                                render={({ field }) => (
+                                    <TextField
+                                        {...field}
+                                        label="Парола"
+                                        id="password"
+                                        name="password"
+                                        margin="normal"
+                                        required
+                                        fullWidth
+                                        type={showPassword ? 'text' : 'password'}
+                                        InputProps={{
+                                            endAdornment: (
+                                                <InputAdornment position="end">
+                                                    <IconButton
+                                                        aria-label="toggle password visibility"
+                                                        onClick={handleClickShowPassword}
+                                                        onMouseDown={handleMouseDownPassword}
+                                                    >
+                                                        {showPassword ? <VisibilityOff /> : <Visibility />}
+                                                    </IconButton>
+                                                </InputAdornment>
+                                            ),
+                                        }}
+                                        autoComplete="current-password"
+                                        color="secondary"
+                                        error={!!errors.password}
+                                        helperText={errors.password ? errors.password.message : ''}
+                                    />
+                                )}
+                            />
+
+                            <Button
+                                type="submit"
+                                fullWidth
+                                variant="contained"
+                                sx={{ mt: 3, mb: 2 }}
+                                disabled={Object.keys(errors).length > 0}
```

Comment:
you can maybe get this out into its own function like `formHasErrors`

Reply:
fixed